api/models.py

```diff
@@ -215,6 +215,19 @@ def set_timeout(self, hours: int = 24):
     def update(self):
         self.updated = datetime.utcnow()
 
+    @classmethod
+    def validate_timestamp(cls, params: dict):
+        """Validate timestamp fields in `params` to be valid isoformat
+        string"""
+        timestamp_fields = ('created', 'updated', 'timeout', 'holdoff')
+        for key in params.keys():
+            if any(field in key for field in timestamp_fields):
```

Comment:
Can we not do something like this instead?
```python
for key, value in params.items():
  if key in timestamp_fields:
    datetime.fromisoformat(value)
```

Reply:
I have dropped `validate_timestamp` method and moved `translate_fields` to try-except block to catch exceptions when invalid isoformat string will be received.

Reply:
However, `if key in timestamp_fields:` will not be able to evaluate correctly when a field is received with an operator e.g. `'created__lt' in ('created', 'updated', 'timeout', 'holdoff')` will be `False`.

Reply:
Ah I see.  Maybe the operators should be parsed first so just the field names appear here, but I'm not sure what's the best way to implement that.

Reply:
Yes, that can be done too. Let me see.

Reply:
Updated it. @gctucker